src/client/datascience/kernel-launcher/remoteKernelFinder.ts

```diff
@@ -55,22 +55,42 @@ export class RemoteKernelFinder implements IRemoteKernelFinder {
     public async findKernel(
         resource: Resource,
         connInfo: INotebookProviderConnection | undefined,
-        option?: nbformat.INotebookMetadata | PythonEnvironment,
+        notebookMetadata?: nbformat.INotebookMetadata,
         _cancelToken?: CancellationToken
     ): Promise<KernelConnectionMetadata | undefined> {
-        // Get list of all of the specs
-        const kernels = await this.listKernels(resource, connInfo);
+        const resourceType = getResourceType(resource);
+        const language =
+            resourceType === 'interactive' ? PYTHON_LANGUAGE : getLanguageInNotebookMetadata(notebookMetadata) || '';
+        try {
+            // Get list of all of the specs
+            const kernels = await this.listKernels(resource, connInfo);
 
-        // Find the preferred kernel index from the list.
-        const notebookMetadata = option && !isInterpreter(option) ? option : undefined;
-        return findPreferredKernel(
-            kernels,
-            resource,
-            [],
-            notebookMetadata,
-            undefined,
-            this.preferredRemoteKernelIdProvider
-        );
+            // Find the preferred kernel index from the list.
+            const preferred = findPreferredKernel(
+                kernels,
+                resource,
+                [],
+                notebookMetadata,
+                undefined,
+                this.preferredRemoteKernelIdProvider
+            );
+            sendTelemetryEvent(Telemetry.PreferredKernel, undefined, {
+                result: preferred ? 'found' : 'notfound',
+                resourceType,
+                language
+            });
+            return preferred;
+        } catch (ex) {
+            sendTelemetryEvent(
```

Comment:
Added a try catch, just as we do today with localKernelFinder
& swallowed exception as we do in localKernelFinder as well.
Also this fuction does support returning `undefined`

Reply:
Thought its a chnage, i think its the right thing to do here, ignore the exception instead of bombing out (jsut as we do today with local kernel finer)